pkg/api/v1beta3/types.go

```diff
@@ -206,7 +206,10 @@ type HostPathVolumeSource struct {
 	Path string `json:"path" description:"path of the directory on the host"`
 }
 
-type EmptyDirVolumeSource struct{}
+type EmptyDirVolumeSource struct {
+	// Optional: make this empty directory be backed by memory.
+	Tmpfs bool `json:"tmpfs" description:"use tmpfs instead of disk"`
```

Comment:
@thockin I don't buy the argument. :-)

I don't want to introduce a deprecated field into v1beta3 that we'll have to carry for more than a year.

Also, for most people the type (tmpfs, ssd) is easier to reason about than qualities. How many people do we actually expect to know what latency and/or throughput ssds can be expected to deliver on a given cluster? FWIW, this is also true for availability. Most people will read a blog about best practices and just conclude they should spread across racks and run N+2, they aren't going to measure what availability they need and/or achieve. We shouldn't force users to reverse-engineer how to achieve what they think they want. 

If/when we add qualities for sophisticated users / use cases, they should be complementary to type. Qualities will probably be another map, like ResourceList (which is now a misnomer), or possibly a list of key, operator, value.